tests/functional/backend/SettingsPluginTest.php

```diff
@@ -75,43 +77,112 @@ public function testGetAndSetSetting()
-            $this->assertEquals($setting->value, json_encode($value) );
+            $this->assertEquals($setting->value, json_encode($value));
 
             $settingValue = self::$plugin->getSetting($key);
             $this->assertEquals($settingValue, json_decode(json_encode($value), true));
         }
     }
 
+    /**
+     * Testing for settings at a suvey level.
+     */
+    public function testGetAndSetSurveySetting()
+    {
+
+        foreach (self::$settings as $key => $value) {
+            self::$plugin->setSurveySetting($key, $value, self::$surveyId);
+
+            $setting = \PluginSetting::model()->findByAttributes([
+                'plugin_id' => self::$plugin->getId(),
+                'model' => 'Survey',
+                'model_id' => self::$surveyId,
+                'key' => $key
+            ]);
+
+            $this->assertNotEmpty($setting->id, 'Setting id should not be empty.');
+            $this->assertEquals(
+                $setting->value,
+                json_encode($value),
+                'The setting value obtained with the PluginSetting model should be the value previously set'
+            );
+
+            $settingValue = self::$plugin->getSurveySetting($key, self::$surveyId);
+            $this->assertEquals(
+                $settingValue,
+                json_decode(json_encode($value), true),
+                'The setting value obtained with the get function should be the value previously set'
+            );
+        }
+    }
+
     public function testGetAndSetSettingEncripted()
     {
         self::$plugin->setEncriptedSettings(array_keys(self::$settings));
         foreach (self::$settings as $key => $value) {
             self::$plugin->setSetting($key, $value);
-        
+
             $setting = \PluginSetting::model()->findByAttributes([
                 'plugin_id' => self::$plugin->getId(),
                 'key' => $key
             ]);
-            
+
             $this->assertNotEmpty($setting->id);
             if (empty($value)) {
-                $this->assertEquals($setting->value, json_encode($value) );
+                $this->assertEquals($setting->value, json_encode($value));
             } else {
-                $this->assertEquals($setting->value, 
-                    json_encode(LSActiveRecord::encryptSingle(json_encode($value)))
-                );
+                $this->assertEquals($setting->value, json_encode(LSActiveRecord::encryptSingle(json_encode($value))));
             }
 
             $settingValue = self::$plugin->getSetting($key);
-            $this->assertEquals($settingValue, json_decode(json_encode($value), true) );
+            $this->assertEquals($settingValue, json_decode(json_encode($value), true));
         }
     }
 
+    /**
+     * Testing for encrypted settings at a survey level
+     */
+    public function testGetAndSetSurveySettingEncripted()
+    {
+        self::$plugin->setEncriptedSettings(array_keys(self::$settings));
```

Comment:
setEncriptedSettings

Spelling error

Reply:
How can the test run correctly when the function is misspelled?

Reply:
It is misspelled both: on th definition and invocation :)

Reply:
OK can we fix that? :joy:

Reply:
Done

Reply:
It was fixed in https://github.com/LimeSurvey/LimeSurvey/pull/2753 too.